packages/strapi-design-system/src/helpers/throwPropError.js

```diff
@@ -0,0 +1,25 @@
+import PropTypes from 'prop-types';
+
+/**
+ * @type {(otherProps: string[]) => (props: Record<string, unknown>, propName: string) => Error | undefined}
+ */
+
+export const throwPropErrorIfNoneAreDefined = (otherProps, propType) => (props, propName, componentName) => {
+  if (!props[propName] && otherProps.every((otherProp) => !props[otherProp])) {
+    return new Error(`One of the following props is required: ${propName}, ${otherProps.join(', ')}`);
+  }
+
+  return PropTypes.checkPropTypes({ [propName]: PropTypes[propType] }, props, 'prop', componentName);
+};
+
+/**
+ * @type {(otherProps: any{}) => (props: Record<string, unknown>, propName: string) => Error | undefined}
+ */
+
+export const throwPropErrorRequiredIf = (otherProps, propType) => (props, propName, componentName) => {
```

Comment:
```suggestion
export const throwPropErrorIfRequired = (otherProps, propType) => (props, propName, componentName) => {
```

Reply:
found that a bit hard to read to begin with?

Reply:
works for me 😊